packages/components/breadcrumbs/breadcrumbs.ts

```diff
@@ -0,0 +1,161 @@
+export const kbqBreadcrumbsConfigurationProvider = (configuration: KbqBreadcrumbsConfiguration): Provider => ({
+    provide: KBQ_BREADCRUMBS_CONFIGURATION,
+    useValue: configuration
+});
+
+@Directive({
+    selector: '[kbqBreadcrumbsSeparator]',
+    standalone: true
+})
+export class KbqBreadcrumbsSeparator {
+    readonly templateRef = inject(TemplateRef);
+}
+
+/**
+ * This directive is used to style breadcrumb buttons with default styling rules.
+ *
+ */
+@Directive({
+    standalone: true,
+    hostDirectives: [RdxRovingFocusItemDirective],
+    selector: '[kbq-button][kbqBreadcrumb]',
+    host: { class: 'kbq-breadcrumb' }
+})
+export class KbqDefaultBreadcrumbStyler implements OnInit {
+    button = inject(KbqButton, { optional: true, host: true });
+
+    ngOnInit() {
+        if (this.button) {
+            this.button.color = KbqComponentColors.Contrast;
+            this.button.kbqStyle = KbqButtonStyles.Transparent;
+        }
+    }
+}
+
+/**
+ * Directive provides a way to define a custom template for breadcrumb rendering, leveraging TemplateRef
+ */
+@Directive({
+    standalone: true,
+    selector: '[kbqBreadcrumbView]'
+})
+export class KbqBreadcrumbView {
+    templateRef = inject(TemplateRef);
+}
+
+/**
+ * Directive represents an individual breadcrumb item with optional support for router navigation and styling.
+ */
+@Directive({
+    standalone: true,
+    selector: 'kbq-breadcrumb-item',
+    host: {
+        class: 'kbq-breadcrumb'
+    }
+})
+export class KbqBreadcrumb {
+    @Input() text: string;
+    @Input({ transform: booleanAttribute }) disabled: boolean;
+
+    @ContentChild(KbqBreadcrumbView, { read: TemplateRef }) customTemplateRef: TemplateRef<any>;
+
+    routerLink = inject(RouterLink, { optional: true, host: true });
+}
+
+@Component({
+    standalone: true,
+    selector: 'kbq-breadcrumbs,[kbq-breadcrumbs]',
+    templateUrl: './breadcrumbs.html',
+    styleUrls: ['./breadcrumbs.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    imports: [
+        AsyncPipe,
+        NgTemplateOutlet,
+        RouterLink,
+        KbqIconModule,
+        KbqButtonModule,
+        KbqDropdownModule,
+        KbqBreadcrumb,
+        KbqDefaultBreadcrumbStyler
+    ],
+    host: {
+        '[class]': 'class',
+        '[attr.aria-label]': "'breadcrumb'"
+    },
+    hostDirectives: [RdxRovingFocusGroupDirective]
+})
+export class KbqBreadcrumbs implements AfterContentInit {
+    protected readonly options = inject(KBQ_BREADCRUMBS_CONFIGURATION);
+    protected readonly destroyRef = inject(DestroyRef);
+    protected readonly cdr = inject(ChangeDetectorRef);
+
+    @Input()
+    size: KbqDefaultSizes = this.options.size;
+
+    @Input()
+    max: number | null = this.options.max;
+
+    @ContentChild(KbqBreadcrumbsSeparator, { read: TemplateRef })
+    separator?: TemplateRef<any>;
+
+    @ContentChildren(KbqBreadcrumb)
+    items: QueryList<KbqBreadcrumb>;
```

Comment:
```suggestion
    protected readonly items: QueryList<KbqBreadcrumb>;
```
используй модификаторы, помогает избежать некоторых ошибок